pact/provider/core/src/main/java/org/arquillian/algeron/pact/provider/core/StateTypeConverter.java

```diff
@@ -0,0 +1,149 @@
+package org.arquillian.algeron.pact.provider.core;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+
+public class StateTypeConverter {
```

Comment:
ok, about the test, and we support primitives, notice that one ftest is using a native